docs/user/setting_up_your_development_environment.md

```diff
@@ -0,0 +1,101 @@
+
+## Overview
+
+Garage is a Python package which can be installed in most Python 3.6+
+environments using standard commands, i.e.
+
+``pip install --user garage``
+
+However, to ensure that dependencies get resolved properly, we suggest
+using a Python environment manager that supports dependency resolution,
+such as [pipenv](https://docs.pipenv.org/en/latest/),
+[conda](https://docs.conda.io/en/latest/), or
+[poetry](https://poetry.eustace.io/). We test against pipenv and conda.
+
+Garage is also tested in [virtualenv](https://virtualenv.pypa.io/en/latest/), but
+virtualenv occasionally has difficulty resolving dependency conflicts, so use with
+care.
+
+_Note: Starting with our 2020.06 release, we only support Python 3.6+._
+
+## Installing Dependencies
+
+Generally speaking, system dependencies of garage are minimal, and likely already
+installed. However, many of the environments garage is used with have additional
+dependencies, and we provide "setup scripts" for installing those dependencies
+and working around known problems on common platforms.
+
+If you can already use the environments you need, you can skip this section.
+
+A MuJoCo key is required for full functionality. You can get one
+[here](https://www.roboti.us/license.html).
+
+### Running the installation scripts
+
+Clone our [repository](https://github.com/rlworkgroup/garage) and navigate to its
+directory. Then run the script like this:
+
+- Linux:
+
+```
+./scripts/setup_linux.sh --mjkey path-to-your-mjkey.txt
+```
+
+- macOS 10.x:
+
+A list of verified macOS versions can be found at the top of the script. Most
+versions of macOS 10 should work, but we can't speak for versions not in the
+list.
+
+```
+./scripts/setup_macos.sh --mjkey path-to-your-mjkey.txt
+```
+
+Also note that this setup script relies on homebrew to install dependencies.
+If you don't already have homebrew, the script will attempt to install it. If
+you do have homebrew, make sure that your user has permission to install things
+with it. Sometimes npm and brew compete for control over folders, and it's easy
+to mess up folder permissions when trying to resolve those conflicts. If you
+run into these issues, a clean install of homebrew usually solves them.
+
+## Installing Garage as an editable package
+
+- pipenv:
+
+```
+cd path/to/garage/repo
+pipenv --three
+# --pre required because garage has some dependencies with verion numbers <1.0
+pipenv install --pre -e '.[mujoco,dm_control]'
+pipenv install --pre -e '.[all,dev]'
+```
+
+- conda:
+
+```
+conda activate myenv
+pip uninstall garage  # To ensure no existing install gets in the way
+cd path/to/garage/repo
+pip install -e '.[mujoco,dm_control]'
+pip install -e '.[all,dev]'
+```
+
+- virtualenv:
+
+```
+source myenv/bin/activate
+pip uninstall garage  # To ensure no existing install gets in the way
+cd path/to/garage/repo
+pip install -e '.[mujoco,dm_control]'
+pip install -e '.[all,dev]'
+```
+
+----
+
+This page was authored by Hayden Shively
```

Comment:
haha i don't think i really did anything for this page -- just cite yourself! But I appreciate it.